src/core/public/application/capabilities/capabilities_service.test.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { InjectedMetadataService } from '../../injected_metadata';
+import { CapabilitiesService } from './capabilities_service';
+import { basePathServiceMock } from '../../base_path/base_path_service.mock';
+
+describe('#start', () => {
+  let mockFetch: jest.Mock<any>;
+  const basePath = basePathServiceMock.createStartContract();
+  basePath.addToPath.mockImplementation(str => str);
+  const injectedMetadata = new InjectedMetadataService({
+    injectedMetadata: {
+      vars: {
+        uiCapabilities: {
+          foo: { feature: true },
+          bar: { feature: true },
+        },
+      },
+    } as any,
+  }).start();
+  const apps = [{ id: 'app1' }, { id: 'app2', capabilities: { app2: { feature: true } } }] as any;
+
+  beforeEach(() => {
+    mockFetch = jest.spyOn(global as any, 'fetch').mockImplementation((endpoint, options) =>
+      Promise.resolve({
+        status: 200,
+        json: () =>
+          Promise.resolve({ capabilities: JSON.parse((options as any).body).capabilities }),
+      })
+    );
+  });
+
+  afterEach(() => {
+    mockFetch.mockRestore();
+  });
+
+  it('calls backend API with merged capabilities', async () => {
+    const service = new CapabilitiesService();
+    await service.start({ apps, basePath, injectedMetadata });
+    expect(mockFetch.mock.calls).toMatchInlineSnapshot(`
+Array [
```

Comment:
Wanted to ask again here why we prefer matching against inline snapshots instead of a deep equal comparison.